legion/legion/k8s/services.py

```diff
@@ -371,13 +404,36 @@ def delete(self, grace_period_seconds=0):
         core_v1api.delete_namespaced_service(name=self.k8s_service.metadata.name, body=body,
                                              namespace=self.namespace)
 
+        retries = int(os.getenv(*legion.config.K8S_API_RETRIES_COUNT))
+        retry_timeout = int(os.getenv(*legion.config.K8S_API_RETRIES_WAIT))
+
+        service_deleted = ensure_function_succeed(
+            lambda: self.check_service_is_deleted(),
+            retries, retry_timeout, boolean_check=True
+        )
+
+        if not service_deleted:
```

Comment:
@kirillmakhonin how do you distinguish 2 different cases 1) service just was not deleted 2)  the call self.check_service_is_deleted() was not completed by some reason related to k8s ?